cmd/check-markdown/hack.go

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"strings"
+
+	bf "gopkg.in/russross/blackfriday.v2"
```

Comment:
I still suspect using the same markup tools that github itself uses to do the check might have made more sense:
https://github.com/github/markup
https://github.com/gjtorikian/commonmarker
https://github.com/github/cmark-gfm

The last one looks to be the core library, which is in C, so not too avant-garde or arduous maybe. c'est la vie.

Reply:
Point already taken but:

- This tool was based on a test tool I had already half-written a while back.

  Although I think there is a lot more this tool could be made to do, I just want the checking integrated into the CI asap as we continue to see doc breakages and this tool will stop them (frankly, I'm surprised GitHub doesn't offer this sort of functionality already).

  Once we have "something" in the CI to perform the checks, we can always replace it with "something else" later.

- markup is ruby and I'm very rusty on that.

- the other 2 packages are written in C which -- although I love it -- is not a language we should probably be writing new code in.
- writing it in go means the rest of the team will understand it (hopefully :-) but we can also leverage all the go goodness we rely on so much (`go test`, `gofmt`, etc).

Of course, if you or anyone else in the community would like to *take up the gauntlet* and rewrite this in ruby/rust/blah, be my guest - I won't be offended! :smile: